components/org.wso2.carbon.identity.conditional.auth.functions.utils/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/utils/FilterAuthenticatorsFunctionImpl.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.conditional.auth.functions.utils;
+
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Implementation of the {@link FilterAuthenticatorsFunction}.
+ */
+public class FilterAuthenticatorsFunctionImpl implements FilterAuthenticatorsFunction {
+    @Override
+    public Map<String, Map<String, String>> filterAuthenticators(List<Map<String, String>> authenticatorOptions,
+                                                                 String excludeAuthenticator) {
+
+        Map<String, Map<String, String>> result = new HashMap<>();
+        int index = 0;
+
+        for (Map<String, String> option : authenticatorOptions) {
+            String idp = option.get(FrameworkConstants.JSAttributes.IDP);
+            String authenticator = option.get(FrameworkConstants.JSAttributes.AUTHENTICATOR);
+
+            if (!excludeAuthenticator.equals(authenticator)) {
+                Map<String, String> idpMap = new HashMap<>();
+                if (FrameworkConstants.LOCAL_IDP_NAME.equals(idp)) {
+                    idpMap.put(FrameworkConstants.JSAttributes.AUTHENTICATOR, authenticator);
+                } else {
+                    idpMap.put(FrameworkConstants.JSAttributes.IDP, idp);
+                }
+                result.put(String.valueOf(index++), idpMap);
```

Comment:
Why can't we pass the integer value as it is here? Why do we need to make this value string?

Reply:
As in https://github.com/wso2/carbon-identity-framework/blob/1ceee196051722474017ab432cee84d4a949b7a3/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/JsNashornGraphBuilder.java#L371C27-L371C61, the expected return format is `Map<String, Map<String, String>>`